tests/tests/image_atomics/mod.rs

```diff
@@ -0,0 +1,168 @@
+//! Tests for image atomics.
+
+use wgpu::ShaderModuleDescriptor;
+use wgpu_test::{gpu_test, GpuTestConfiguration, TestParameters, TestingContext};
+
+#[gpu_test]
+static IMAGE_32_ATOMICS: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .limits(wgt::Limits {
+                max_storage_textures_per_shader_stage: 1,
+                max_compute_invocations_per_workgroup: 64,
+                max_compute_workgroup_size_x: 4,
+                max_compute_workgroup_size_y: 4,
+                max_compute_workgroup_size_z: 4,
+                max_compute_workgroups_per_dimension: wgt::COPY_BYTES_PER_ROW_ALIGNMENT,
+                ..wgt::Limits::downlevel_webgl2_defaults()
+            })
+            .features(
+                wgpu::Features::TEXTURE_ATOMIC
+                    | wgpu::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES,
+            ),
+    )
+    .run_async(|ctx| async move {
+        test_format(
+            ctx,
+            wgpu::TextureFormat::R32Uint,
+            wgpu::include_wgsl!("image_32_atomics.wgsl"),
+        )
+        .await;
+    });
+
+async fn test_format(
+    ctx: TestingContext,
+    format: wgpu::TextureFormat,
+    desc: ShaderModuleDescriptor<'_>,
+) {
+    let pixel_bytes = format.target_pixel_byte_cost().unwrap();
+    let size = wgpu::Extent3d {
+        width: wgt::COPY_BYTES_PER_ROW_ALIGNMENT,
+        height: wgt::COPY_BYTES_PER_ROW_ALIGNMENT,
+        depth_or_array_layers: 1,
+    };
+    let bind_group_layout_entries = vec![wgpu::BindGroupLayoutEntry {
+        binding: 0,
+        visibility: wgpu::ShaderStages::COMPUTE,
+        ty: wgpu::BindingType::StorageTexture {
+            access: wgpu::StorageTextureAccess::Atomic,
+            format,
+            view_dimension: wgpu::TextureViewDimension::D2,
+        },
+        count: None,
+    }];
+
+    let bind_group_layout = ctx
+        .device
+        .create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+            label: None,
+            entries: &bind_group_layout_entries,
+        });
```

Comment:
```suggestion
    let bind_group_layout_entry = wgpu::BindGroupLayoutEntry {
        binding: 0,
        visibility: wgpu::ShaderStages::COMPUTE,
        ty: wgpu::BindingType::StorageTexture {
            access: wgpu::StorageTextureAccess::Atomic,
            format,
            view_dimension: wgpu::TextureViewDimension::D2,
        },
        count: None,
    };

    let bind_group_layout = ctx
        .device
        .create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
            label: None,
            entries: std::slice::from_ref(&bind_group_layout_entry),
        });
```

Reply:
why not `&[bind_group_layout_entry]` instead of `std::slice::from_ref`?

Reply:
You can do that too, 6 of one